Generalize interfaces initializer

nilsvu opened the pull request:
## Proposed changes

Gives the interfaces initializer the same treatment as the domain initializer in #1612 so that it can be used by elliptic systems.

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsvu committed:
Move interface initializer to DiscontinuousGalerkin

nilsvu commented:
@osheamonn could you review this as well? If you don't have time, please feel free to decline / forward the review to someone else. For the second reviewer, perhaps @fmahebert could you review?

osheamonn commented on src/ParallelAlgorithms/DiscontinuousGalerkin/InitializeInterfaces.hpp:
```diff
@@ -103,6 +109,8 @@ struct InitializeInterfaces {
       ::Tags::InterfaceComputeItem<Directions, ::Tags::Direction<dim>>,
       ::Tags::InterfaceComputeItem<Directions, ::Tags::InterfaceMesh<dim>>,
+      ::Tags::InterfaceComputeItem<
+          Directions, ::Tags::BoundaryCoordinates<dim, Frame::Inertial>>,
```
My only question is why are the coordinates on the internal faces being added? The coordinates for the external faces are needed for imposing boundary conditions but I'm not familiar with a use case for the coordinates on the internal faces.

nilsvu replied:
Turns out I don't need them after all, I should use the exterior boundary coordinates instead. Thanks for pointing this out. Removed this.

I also removed the part where I move the exterior variables simple tag to the Evolution/DG initializer, because this needs further thought in a future PR (For systems where the metric is a variable we need it on exterior faces to compute the FaceNormal. I know what to do about it, but it's not part of this PR and makes this one easier). Since this is a lot of removed code I squashed it directly, I think it's easier to review like this.

nilsvu committed:
Generalize interfaces initializer

Cleans up code and documentation and adds a test.
This initializer will be used by both the evolution and elliptic code.

nilsvu commented:
@osheamonn Are your issues resolved?

@fmahebert could you review? Just let me know so that I find another reviewer if you don't have time.

fmahebert commented:
@nilsleiffischer I am happy to review, though it may take 2-3 days for me to get there.

nilsdeppe commented:
@osheamonn could you please let us know if you're happy with this or would like to see more changes? Thanks!

nilsvu commented:
@sxs-collaboration/spectre-core-devs this is ready to be merged. Thanks for reviewing @osheamonn @nilsdeppe
